solidity/random-beacon/test/RandomBeacon.Authorization.test.ts

```diff
@@ -0,0 +1,2949 @@
+                randomBeacon,
+                "InvoluntaryAuthorizationDecreaseFailed"
+              )
+            })
+          })
+
+          context(
+            "when the authorized amount drops to below the minimum",
+            () => {
+              before(async () => {
+                const slashingTo = minimumAuthorization.sub(1)
+                const slashingBy = stakedAmount.sub(slashingTo)
+
+                await createSnapshot()
+
+                // slash!
+                await staking
+                  .connect(slasher.wallet)
+                  .slash(slashingBy, [stakingProvider.address])
+
+                await staking.connect(thirdParty).processSlashing(1)
+              })
+
+              after(async () => {
+                await restoreSnapshot()
+              })
+
+              it("should remove operator from the sortition pool", async () => {
+                expect(await randomBeacon.isOperatorInPool(operator.address)).to
+                  .be.false
+              })
+            }
+          )
+        })
+      })
+    })
+  })
+
+  describe("joinSortitionPool", () => {
+    context("when the operator is unknown", () => {
+      it("should revert", async () => {
+        await expect(
+          randomBeacon.connect(thirdParty).joinSortitionPool()
+        ).to.be.revertedWith("Unknown operator")
+      })
+    })
+
+    context("when the operator has no stake authorized", () => {
+      before(async () => {
+        await createSnapshot()
+
+        await randomBeacon
+          .connect(stakingProvider)
+          .registerOperator(operator.address)
+      })
+
+      after(async () => {
+        await restoreSnapshot()
+      })
+
+      it("should revert", async () => {
+        await expect(
+          randomBeacon.connect(operator).joinSortitionPool()
+        ).to.be.revertedWith("Authorization below the minimum")
+      })
+    })
+
+    // The only option for it to happen is when there was a slashing.
+    context(
+      "when the authorization dropped below the minimum but is still non-zero",
+      () => {
+        before(async () => {
+          await createSnapshot()
+
+          await randomBeacon
+            .connect(stakingProvider)
+            .registerOperator(operator.address)
+
+          const authorizedAmount = minimumAuthorization
+          await staking
+            .connect(authorizer)
+            .increaseAuthorization(
+              stakingProvider.address,
+              randomBeacon.address,
+              authorizedAmount
+            )
+
+          const slashingTo = minimumAuthorization.sub(1)
+          // Note that we slash from the entire staked amount given that the
+          // initially authorized amount is less than staked amount and it is
+          // another application slashing. To go below the minimum stake, we need
+          // to start slashing from the entire staked amount, not just the
+          // one authorized for WalletRegistry.
```

Comment:
```suggestion
          // one authorized for RandomBeacon.
```

There are more occurrences of `WalletRegistry` in comments in this file.

Reply:
f931643f8c9330c677ed937b457aed7e45072fe9